### DISCUSSION TITLE
Is there a way to add a room location string to the tasmota/disovery/+/config ?

### DISCUSSION THREAD

**Discussion by sprior** [arendst/Tasmota#20145] [Q&A]
I'm starting to incorporate Tasmota into a HA system I've written from scratch.  The model my older devices work under is that you tell the device what it's location in the house is and the device includes that info in its discovery broadcast on boot.  This info is parsed and loaded into Redis for all systems to use.  I'm looking at the payload of the tasmota/disovery/+/config topic and don't see anything that would obviously be the location/room of the device.  Is there a place where I could put this in the existing structure?

**Comment by barbudor** [comment]
For me the most obvious way is to use the topic structure by adapting fhe FullTopic
Fulltopic default value is `%prefix%/%topic%`
You can insert in that structure information of your liking
For example `%prefix%/floor-1/room-2/%topic%` for all devices in room2 of floor1.
If you structure your full topics that was, your home-automation system will be able to extract that information.

**Reply by sfromis** [reply]
To quote myself from Discord:

You could include something in one of the name strings included, like `DeviceName`, template module name, or one of the `FriendlyName` or `SwitchText` values.

Or you could "invent" a new retained payload, using a rule to have the device publish it at boot time.
